theme/js/dataset-recommendations.js

```diff
@@ -0,0 +1,67 @@
+const recoUrl = 'https://raw.githubusercontent.com/etalab/datasets_reco/master/reco_json/';
+const maxRecos = 2;
+
+function getDatasetId() {
+    const selector = '#json_ld';
+    const dataset = JSON.parse(document.querySelector(selector).text);
+    return dataset && dataset['@id'];
+}
+
+function addRecos(recos) {
+    const recoContainer = document.getElementById('dataset-recommendations-container');
+    let recoChildContainer, recoChildEmbed;
+    recos.splice(0, maxRecos).forEach((reco) => {
+        recoChildContainer = document.createElement('div');
+        recoChildContainer.classList.add('recommendation');
+    
+        recoChildEmbed = document.createElement('div');
+        recoChildEmbed.setAttribute('data-udata-dataset-id', reco[0]);
+    
+        recoChildContainer.appendChild(recoChildEmbed);
+        recoContainer.appendChild(recoChildContainer);
+    });
+}
+
+function addWidgetScript() {
```

Comment:
Why not add the script tag in the [theme template](https://github.com/etalab/udata-gouvfr/blob/master/udata_gouvfr/theme/templates/dataset/display.html) ?

Reply:
Because we do not load the script if there is no recommendation. Quite an optimization IMHO since we have recommendations for the top 30 or 50 datasets only.